### PR TITLE
chore(examples): cli args

### LINKED ISSUE #438: Allow CLI args in some examples
## Description

There are some examples that don't work without the user changing a constant. I think the better approach would be to make this a CLI argument.

## Motivation

DX

## Are you planning to do it yourself in a pull request?

Sure.

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Some examples depend on other examples by relying on results of those. This PR changes those examples to receive those results as command line arguments.

## Links to any relevant issues

Closes #438 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Unit tests.

### CODE REVIEW

**Comment by thibault-martinez** [comment]
There was only one such example?

File: sdk/examples/wallet/13_burn_native_token.rs [hunk 1]
```diff
@@ -8,47 +8,46 @@
 //! Make sure that `example.stronghold` and `example.walletdb` already exist by
 //! running the `create_account` example!
 //!
+//! You need to provide a TOKEN ID that is available in the account. The foundry
+//! output which minted it needs to be available as well. You can check this by
```
Comment (Thoralf-M) [inline sdk/examples/wallet/13_burn_native_token.rs]: Foundry is not needed for burning

Comment (Alex6323) [reply sdk/examples/wallet/13_burn_native_token.rs]: I am pretty sure I didn't write this. Just moved it around. But will remove it :+1:

Comment (Thoralf-M) [reply sdk/examples/wallet/13_burn_native_token.rs]: Well the foundry output needs to be available for increase/decrease, but not for burning :trollface:

Comment (Alex6323) [reply sdk/examples/wallet/13_burn_native_token.rs]: ooh, copy pasta :spaghetti: you got me!

File: sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs [hunk 2]
```diff
@@ -4,15 +4,17 @@
...
-//! by running the `mint_issuer_nft` example!
+//! running the `create_account` example.
+//!
+//! You have to provide the ISSUER NFT ID that you created by running the
+//! `mint_issuer_nft` example!
 //!
 //! Rename `.env.example` to `.env` first, then run the command:
 //! ```sh
-//! cargo run --release --all-features --example mint_collection_nft
+//! cargo run --release --all-features --example mint_collection_nft [ISSUER_NFT_ID]
```
Comment (thibault-martinez) [inline sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs]: Don't put square brackets if they are mandatory

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs (hunk 2 above)
Comment (Alex6323) [reply sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs]: good catch ... fixed

File: sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs [hunk 3]
```diff
@@ -4,15 +4,17 @@
...
+//! `mint_issuer_nft` example!
 //!
 //! Rename `.env.example` to `.env` first, then run the command:
 //! ```sh
-//! cargo run --release --all-features --example mint_collection_nft
+//! cargo run --release --all-features --example mint_collection_nft <ISSUER NFT ID>
 //! ```
 
-use std::{env::var, str::FromStr};
+use std::env;
```
Comment (thibault-martinez) [inline sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs]: Do the thing here as well then

**Comment by kwek20** [comment]
Should also do this in the bindings examples right?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

File: sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs (hunk 3 above)
Comment (Alex6323) [reply sdk/examples/wallet/nft_collection/01_mint_collection_nft.rs]: Done

File: sdk/examples/how_tos/native_tokens/burn.rs [hunk 4]
```diff
@@ -8,14 +8,19 @@
 //! Make sure that `example.stronghold` and `example.walletdb` already exist by
 //! running the `create_account` example!
 //!
+//! You may provide a TOKEN ID that is available in the account. You can check this by running the
+//! `get_balance` example. You can mint a new native token by running the `mint_native_token` example.
+//!
 //! Rename `.env.example` to `.env` first, then run the command:
 //! ```sh
-//! cargo run --release --all-features --example burn_native_token
+//! cargo run --release --all-features --example burn_native_token [TOKEN ID]
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/native_tokens/burn.rs]: For this and the others, please put underscores in the param name: `TOKEN_ID`

Comment (Alex6323) [reply sdk/examples/how_tos/native_tokens/burn.rs]: Done

File: sdk/examples/how_tos/native_tokens/burn.rs [hunk 5]
```diff
@@ -37,52 +42,53 @@ async fn main() -> Result<()> {
...
+    // Take the given token id, or use a default.
+    let token_id = std::env::args()
+        .nth(1)
+        .map(|s| s.parse::<TokenId>().expect("invalid token id"))
+        .unwrap_or_else(|| TokenId::from(*balance.foundries().first().unwrap()));
+
+    if let Some(native_token_balance) = balance.native_tokens().iter().find(|native_token| {
+        native_token.token_id() == &token_id && native_token.available() >= U256::from(MIN_AVAILABLE_AMOUNT)
+    }) {
+        println!("Balance before burning:\n{native_token_balance:#?}",);
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/native_tokens/burn.rs]: Remove

File: sdk/examples/how_tos/native_tokens/burn.rs [hunk 6]
```diff
@@ -37,52 +42,53 @@ async fn main() -> Result<()> {
...
 
         // Set the stronghold password
         wallet
-            .set_stronghold_password(var("STRONGHOLD_PASSWORD").unwrap())
+            .set_stronghold_password(std::env::var("STRONGHOLD_PASSWORD").unwrap())
             .await?;
 
+        println!("Sending the burning transaction...");
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/native_tokens/burn.rs]: Remove

File: sdk/examples/how_tos/native_tokens/burn.rs (hunk 5 above)
Comment (Alex6323) [reply sdk/examples/how_tos/native_tokens/burn.rs]: but this log is in the `.mdx`?

File: sdk/examples/how_tos/native_tokens/burn.rs (hunk 6 above)
Comment (Alex6323) [reply sdk/examples/how_tos/native_tokens/burn.rs]: Done.

File: sdk/examples/how_tos/native_tokens/burn.rs (hunk 5 above)
Comment (Alex6323) [reply sdk/examples/how_tos/native_tokens/burn.rs]: made the output look like the example given in `burn.mdx`

**Review by Thoralf-M — APPROVED** [review]

File: sdk/examples/how_tos/native_tokens/burn.rs (hunk 5 above)
Comment (thibault-martinez) [reply sdk/examples/how_tos/native_tokens/burn.rs]: Ah my bad then, though you were just adding it

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**